File: easy/344_reverse_string/go/reverse_string.go
package leetcode

func reverseString(s []byte) {
	p1 := 0
	p2 := len(s) - 1
	for p1 < p2 {
		s[p1], s[p2] = s[p2], s[p1]
		p1 = p1 + 1
		p2 = p2 - 1
	}
}

File: easy/344_reverse_string/java/app/src/main/java/solution/ReverseString.java
package solution;

public class ReverseString {
    public void reverseString(char[] s) {
        int p1 = 0;
        int p2 = s.length - 1;
        while (p1 < p2) {
            char tmp;
            tmp = s[p1];
            s[p1] = s[p2];
            s[p2] = tmp;
            p1++;
            p2--;
        }
    }
}